addons/snailmail/models/ir_actions_report.py

```diff
@@ -6,6 +6,14 @@
 class IrActionsReport(models.Model):
     _inherit = 'ir.actions.report'
 
+    @api.multi
+    def is_using_attachment(self):
```

Comment:
Could we try to do the same kind of trick directly in `retrieve_attachment()`: returning nothing when `self.env.context.get('snailmail_layout')` should give the same result. This way we don't need to introduce a new report method in 12.0.

Also it seems that we don't need to alter the layout if the admin has enabled the "cover page" feature. In that case it would be better to keep the normal behavior and use the real document if it exists (i.e. the "official" or "legal" PDF), and avoid the special case here.

Reply:
Ok, I made the change.
For the cover page it was only introduced in 12.2, we will make the modification in that version.